src/main/kotlin/eli/inevitable/formatters/CellFormatter.kt

```diff
@@ -0,0 +1,261 @@
+package eli.inevitable.formatters
+
+import eli.inevitable.enumerations.Align
+import eli.inevitable.formatters.TextFormatter.Companion.LINE_FORMAT
+import eli.inevitable.isPrintable
+import eli.inevitable.maxWidth
+import eli.inevitable.sanitize
+import eli.inevitable.templates.AbstractBuilderTemplate
+
+/**
+ * A generalized formatter that constructs a block of formatted text. This block may optionally be padded with an arbitrary number of empty rows of text.
+ */
+class CellFormatter private constructor(): TextFormatter {
+
+    class Builder: AbstractBuilderTemplate<CellFormatter> {
+
+        constructor(formatter: CellFormatter): super(CellFormatter()) {
+            buildable.run {
+                rawLines.addAll(formatter.rawLines)
+                alignment = formatter.alignment
+                fillingSpacerToken = formatter.fillingSpacerToken
+                paddingSpacerToken = formatter.paddingSpacerToken
+                verticalPaddingCount = formatter.verticalPaddingCount
+                horizontalPaddingCount = formatter.horizontalPaddingCount
+                lineWidth = formatter.lineWidth
+                height = formatter.height
+                width = formatter.width
+            }
+        }
+
+        constructor(): super(CellFormatter())
+
+        constructor(vararg contents: String): this(contents.toList())
+
+        constructor(contents: List<String>): this() {
+            addLines(contents)
+        }
+
+        companion object {
+            @JvmStatic
+            fun of(original: CellFormatter): Builder = Builder(original)
+
+            @JvmStatic
+            fun of(): Builder = Builder()
+
+            @JvmStatic
+            fun of(vararg contents: String): Builder = Builder(contents.toList())
+
+            @JvmStatic
+            fun of(contents: List<String>): Builder = Builder(contents)
+        }
+
+        /**
+         * @param lines The [array of text lines][String] to populate the rows of the cell.
+         * @return This [Builder]
+         */
+        fun withLines(vararg lines: String): Builder {
```

Comment:
could remove `with` and make it feel more fluent.

Reply:
Agreed. Updated and fixed. Will apply these changes to the other Formatters as well.